inference-engine/src/legacy_api/src/convert_function_to_cnn_network.cpp

```diff
@@ -401,7 +401,7 @@ void CNNLayerCreator::on_adapter(const std::string& name,
         params[name] = details::convertPrecision(type).name();
     } else if (auto a = ::ngraph::as_type<::ngraph::AttributeAdapter<::ngraph::PartialShape>>(&adapter)) {
         std::string dims;
-        auto shape = static_cast<::ngraph::PartialShape&>(*a);
+        auto shape = a->get();
```

Comment:
Can the `shape` have a dynamic rank?

Reply:
Yes, partial shape may have a dynamic rank, PartialShape class allows it.

Reply:
But if `shape` has a dynamic rank, the next line `for (int64_t i = 0; i < shape.rank().get_length(); i++)` is not correct.

Reply:
Yes, you are right. Added rank check.